src/lib/libfetch.js

```diff
@@ -38,7 +38,245 @@ var LibraryFetch = {
+          this.timeout
+        );
+      }
+
+      const fetchOptions = {
+        method: this._method,
+        headers: this._headers,
+        body: body,
+        signal: signal,
+        credentials: this.withCredentials ? 'include' : 'same-origin',
+      };
+
+      try {
+        const response = await fetch(this._url, fetchOptions);
+
+        // Populate response properties once headers are received
+        this.status = response.status;
+        this.statusText = response.statusText;
+        this.responseURL = response.url;
+        this._responseHeaders = response.headers;
+        this._changeReadyState(2); // 2: HEADERS_RECEIVED
+
+        // Start processing the body
+        this._changeReadyState(3); // 3: LOADING
+
+        if (!response.body) {
+          throw new Error("Response has no body to read.");
+        }
+
+        const reader = response.body.getReader();
+        const contentLength = +response.headers.get('Content-Length');
+
+        let receivedLength = 0;
+        const chunks = [];
+
+        while (true) {
+          const { done, value } = await reader.read();
+          if (done) {
+            break;
+          }
+
+          chunks.push(value);
+          receivedLength += value.length;
+
+          if (this.onprogress) {
+            // Convert to ArrayBuffer as requested by responseType.
+            this.response = value.buffer;
+            const progressEvent = {
+              lengthComputable: contentLength > 0,
+              loaded: receivedLength,
+              total: contentLength
+            };
+            this.onprogress(progressEvent);
+          }
+        }
+
+        // Combine chunks into a single Uint8Array.
+        const allChunks = new Uint8Array(receivedLength);
+        let position = 0;
+        for (const chunk of chunks) {
+          allChunks.set(chunk, position);
+          position += chunk.length;
+        }
+
+        // Convert to ArrayBuffer as requested by responseType
+        this.response = allChunks.buffer;
+      } catch (error) {
+        this.statusText = error.message;
+
+        if (error.name === 'AbortError') {
+          // Do nothing.
+        } else if (error.name === 'TimeoutError') {
+          this.ontimeout?.();
+        } else {
+          // This is a network error
+          this.onerror?.();
+        }
+      } finally {
+        clearTimeout(timeoutID);
+        if (!this._aborted) {
+          this._changeReadyState(4); // 4: DONE
+          // The XHR 'load' event fires for successful HTTP statuses (2xx) as well as
+          // unsuccessful ones (4xx, 5xx). The 'error' event is for network failures.
+          this.onload?.();
+        }
+      }
+    }
+
+    /**
+     * Aborts the request if it has already been sent.
+     */
+    abort() {
+      this._aborted = true;
+      this.status = 0;
+      this._changeReadyState(4); // 4: DONE
+      this._abortController?.abort();
+    }
+  }
```

Comment:
What do you think about putting this whole class in `Fetch.js` (you can still wrap it in `#if FETCH_STREAMING`).   I think the idea with this `libfetch.js` seems to be that it just exposes the implementation which lives in `Fetch.js`.

I don't feel strongly about this though.. what do you think?